pkg/extension/sumologicextension/extension.go

```diff
@@ -34,30 +35,37 @@ type sumologicExtension struct {
 	conf             *Config
 	logger           *zap.Logger
 	registrationInfo OpenRegisterResponsePayload
+	ch               chan struct{}
 }
 
 const (
 	// TODO: fix
-	niteBaseUrl = "https://nite-open-events.sumologic.net/"
+	niteBaseUrl              = "https://nite-open-events.sumologic.net/"
+	heartbeatUrl             = "api/v1/collector/heartbeat"
+	registerUrl              = "api/v1/collector/register"
+	defaultHeartbeatInterval = time.Duration(15)
 )
 
 func newSumologicExtension(conf *Config, logger *zap.Logger) (*sumologicExtension, error) {
 	if conf.CollectorName == "" {
 		return nil, errors.New("collector name is unset")
 	}
+	ch := make(chan struct{})
 
 	return &sumologicExtension{
 		// TODO: don't hardcode
 		baseUrl: niteBaseUrl,
 		conf:    conf,
 		logger:  logger,
+		ch:      ch,
```

Comment:
BTW: I'd name this channel differently, e.g. `closeChan`?

```suggestion
	return &sumologicExtension{
		// TODO: don't hardcode
		baseUrl: niteBaseUrl,
		conf:    conf,
		logger:  logger,
		ch:      make(chan struct{}),
```